posts/seurat-cosmx-basics/index.qmd

```diff
@@ -0,0 +1,244 @@
+             cells = row.names(seu.obj@meta.data)[which(seu.obj@meta.data$fov == 99)])
+
+```
+
+```{r}
+#| eval: true
+#| echo: false
+
+knitr::include_graphics("./figures/plot_FOVleiden-1.png")
+```
+
+To color cells by a continuous value, such as the log10totalcounts, or by the expression of a transcript of interest, such as Slc17a7, we use the function 'ImageFeaturePlot'.
+```{r plot_imageContinuousVariable, eval = FALSE}
+# Color cells by log10totalcounts
+ImageFeaturePlot(seu.obj,
+             fov = Images(seu.obj)[1],
+             border.color = NA,
+            features = "log10totalcounts")
+```
+
+```{r}
+#| eval: true
+#| echo: false
+
+knitr::include_graphics("./figures/plot_imageContinuousVariable-1.png")
+```
+
+```{r plot_imageContinuousVariable_gene, eval = FALSE}
+# Color cells by the expression of a gene of interest, Slc17a7
+ImageFeaturePlot(seu.obj,
+             fov = Images(seu.obj)[1],
+             border.color = NA,
+            features = "Slc17a7")
+```
+
+```{r}
+#| eval: true
+#| echo: false
+
+knitr::include_graphics("./figures/plot_imageContinuousVariable_gene-1.png")
+```
+
+Seurat can plot cells with either cell shapes shown ('segmentation') or with a single point at the center of where they're located ('centroids'). Here we show the switch to plotting centroids for one FOV.
+```{r plot_centroids, eval = FALSE}
+# Check what the current default boundary is
+DefaultBoundary(seu.obj@images[[Images(seu.obj)[1]]])
+
+# Change the default boundaries for the first slide
+DefaultBoundary(seu.obj@images[[Images(seu.obj)[1]]]) <- "centroids"
+
+# Plot one FOV from this slide. Note that cell shapes are no longer shown
+ImageDimPlot(seu.obj,
+             fov = Images(seu.obj)[1],
+             size = 5,
+             shuffle.cols = TRUE, # Option to randomly shuffle colors within the palette
+             cells = row.names(seu.obj@meta.data)[which(seu.obj@meta.data$fov == 99)])
+```
+
+```{r}
+#| eval: true
+#| echo: false
+
+knitr::include_graphics("./figures/plot_centroids-1.png")
+```
+
+# Dimension reduction plots {#sec-dim-reduction-plots}
+The CosMx Seurat object contains coordinates for each cell for UMAP dimensional reduction.
+
+Here, we color cells by cell type and overlay cell type labels.
+```{r plot_umapCellTypes, eval = FALSE}
+DimPlot(seu.obj, 
+        group.by = "nb_clus",
+        label = TRUE) +
+  theme(legend.position = "none") # Suppress the legend since labels are plotted on top of UMAP
+```
+
+```{r}
+#| eval: true
+#| echo: false
+
+knitr::include_graphics("./figures/plot_umapCellTypes-1.png")
+```
+
+Here, we color cells by a continuous value, using transcript expression for a transcript of interest.
+```{r plot_umapTranscriptExpression, eval = FALSE}
+FeaturePlot(seu.obj, 
+        features = "Slc17a7",
+        order = TRUE) # plots cells in order of expression
+```
+
+```{r}
+#| eval: true
+#| echo: false
+
+knitr::include_graphics("./figures/plot_umapTranscriptExpression-1.png")
+```
+
+
```

Comment:
Let's write a conclusion section and where we might go next.

Reply:
Good idea. I added the following in commit [e08e360](https://github.com/Nanostring-Biostats/CosMx-Analysis-Scratch-Space/pull/29/commits/e08e360258be181dfc1ecf2de373798c78068013):
"This vignette serves as an introduction to exploring CosMx data in Seurat, with a primary focus on visualization. Mix and match the functions and options from above to generate new customized visualizations with your data. Once you’re comfortable visualizing your spatial data, you may want to proceed to [refining your cell typing](https://nanostring-biostats.github.io/CosMx-Analysis-Scratch-Space/posts/cell-typing-basics/), [performing differential expression](https://nanostring-biostats.github.io/CosMx-Analysis-Scratch-Space/posts/segmentation-error/), [finding spatially correlated genes](https://github.com/Nanostring-Biostats/InSituCor), or countless other analysis paths."